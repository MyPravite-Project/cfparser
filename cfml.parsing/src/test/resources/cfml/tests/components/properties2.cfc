/**
* @author 
*/
component accessors="true"{
  
  // The first name of the User
  property string firstName;
  property lastName;

}