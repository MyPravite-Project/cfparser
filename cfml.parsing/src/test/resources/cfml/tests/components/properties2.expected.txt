/*===TOKENS===*/
Hidden:ML_COMMENT               </**
* @author 
*/>
Hidden:NEWLINE                  <>
COMPONENT                <component>
IDENTIFIER               <accessors>
'='                      <=>
OPEN_STRING              <">
STRING_LITERAL           <true>
CLOSE_STRING             <">
'{'                      <{>
Hidden:NEWLINE                  <>
Hidden:NEWLINE                  <>
Hidden:LINE_COMMENT             <// The first name of the User>
PROPERTY                 <property>
STRING                   <string>
IDENTIFIER               <firstName>
';'                      <;>
Hidden:NEWLINE                  <>
PROPERTY                 <property>
STRING                   <string>
IDENTIFIER               <middleName>
';'                      <;>
Hidden:NEWLINE                  <>
PROPERTY                 <property>
IDENTIFIER               <lastName>
';'                      <;>
Hidden:NEWLINE                  <>
'}'                      <}>
/*===TREE===*/
(scriptBlock
  (componentDeclaration
    component
    (componentAttribute
      (identifier accessors)
      =
      (startExpression
        (compareExpression
          (baseExpression
            (unaryExpression
              (primaryExpression
                (literalExpression (stringLiteral " (stringLiteralPart true) "))
              )
            )
          )
        )
      )
    )
    (componentGuts
      {
      (element
        (statement
          (tagOperatorStatement
            (propertyStatement
              property
              (typeSpec (type string))
              (multipartIdentifier (identifier firstName))
              ;
            )
          )
        )
      )
      (element
        (statement
          (tagOperatorStatement
            (propertyStatement
              property
              (typeSpec (type string))
              (multipartIdentifier (identifier middleName))
              ;
            )
          )
        )
      )
      (element
        (statement
          (tagOperatorStatement
            (propertyStatement property (multipartIdentifier (identifier lastName)) ;)
          )
        )
      )
      }
    )
  )
)
/*======*/
/*===DECOMPILE===*/
component accessors='true' {
  {
property string firstName;
property string middleName;
property lastName;

  }
}
/*======*/